### PR TITLE
minetest-mapserver: init at 4.7.0

### PR DESCRIPTION
**Pull request by zeuner**
## Description of changes

This package provides a web-based real-time mapserver for minetest voxel game servers: https://github.com/minetest-mapserver/mapserver/blob/master/readme.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by cafkafk — COMMENTED** [review]
code lgtm, sending result of nixpkgs-review when done.

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 259915` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>minetest-mapserver</li>
  </ul>
</details>

**Review by cafkafk — APPROVED** [review]
seems to build fine, didn't test with a `world.mt`

**Comment by zeuner** [comment]
> seems to build fine, didn't test with a `world.mt`

Works for me. The binary must be called from the directory where `world.mt` resides.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6424,6 +6424,10 @@
     githubId = 1447245;
     name = "Robin Gloster";
   };
+  gm6k = {
+    email = "nix@quidecco.pl";
```
Comment (nlewo) [inline maintainers/maintainer-list.nix:6428]: Could you add your github nickname also? (could be convenient for automation and other to find you on github)

File: pkgs/by-name/mi/minetest-mapserver/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}:
+
+buildGoModule rec {
+  pname = "minetest-mapserver";
+  version = "4.7.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = "mapserver";
+    rev = "refs/tags/v${version}";
```
Comment (nlewo) [inline pkgs/by-name/mi/minetest-mapserver/package.nix:13]:
```suggestion
    rev = "v${version}";
```

**Review by nlewo — CHANGES_REQUESTED** [review]
Otherwise, lgtm.

File: pkgs/by-name/mi/minetest-mapserver/package.nix (hunk 2 above)
Comment (zeuner) [reply pkgs/by-name/mi/minetest-mapserver/package.nix]: Ok, this also works.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (nlewo) [reply maintainers/maintainer-list.nix:6428]: hehe, I unrderstand!

**Review by nlewo — CHANGES_REQUESTED** [review]
Could you squash your lastest commit with the commit introducing the software (in order to only have 2 commits)?

**Comment by zeuner** [comment]
> Could you squash your lastest commit with the commit introducing the software (in order to only have 2 commits)?

Of course!

**Review by nlewo — APPROVED** [review]
Thank you.

**Pull request merged**